src/pilot/config_idrac.py

```diff
@@ -129,6 +133,13 @@ def configure_nics_boot_settings(
                                                  pxe_nic_id)
 
 
+def get_nic_mac_address(drac_client, pxe_nic_id):
+    for nic in drac_client.list_nics(sort=True):
+        # Compare the NIC IDs case insensitively. Assume ASCII strings.
+        if nic.id.lower() == pxe_nic_id.lower():
+            return nic.mac_address.lower()
```

Comment:
It doesn't look like the above could work because the method that constructs a NIC in python-dracclient has a mac attribute, not a mac_address attribute:
https://github.com/openstack/python-dracclient/blob/master/dracclient/resources/inventory.py#L181